runtime/src/enclave_rpc/session.rs

```diff
@@ -248,10 +281,56 @@ impl Session {
         matches!(self.state, State::Transport(_))
     }
 
+    /// Whether the session is connected to one of the given nodes.
+    pub fn is_connected_to(&self, nodes: &Vec<signature::PublicKey>) -> bool {
+        nodes.iter().any(|&node| Some(node) == self.remote_node)
+    }
+
     /// Whether the session is in closed state.
     pub fn is_closed(&self) -> bool {
         matches!(self.state, State::Closed)
     }
+
+    /// Return remote node identifier.
+    pub fn get_node(&self) -> Result<signature::PublicKey> {
+        self.remote_node.ok_or(SessionError::NodeNotSet.into())
+    }
+
+    /// Set the remote node identifier.
+    pub fn set_remote_node(&mut self, node: signature::PublicKey) -> Result<()> {
+        if self.remote_node.is_some() {
+            return Err(SessionError::NodeAlreadySet.into());
+        }
+        self.remote_node = Some(node);
+        Ok(())
+    }
+
+    fn node_to_rak(&self) -> Result<Option<signature::PublicKey>> {
+        let consensus_verifier = self
+            .consensus_verifier
+            .as_ref()
+            .expect("consensus verifier should be set");
+        let runtime_id = self.remote_runtime_id.ok_or(SessionError::RuntimeNotSet)?;
+        let node = self.remote_node.ok_or(SessionError::NodeNotSet)?;
+
+        let consensus_state = consensus_verifier.latest_state()?;
+        let registry_state = RegistryState::new(&consensus_state);
+        let node = registry_state
+            .node(Context::background(), &node)?
+            .ok_or(SessionError::NodeNotRegistered)?;
+        let runtimes = node.runtimes.unwrap_or_default();
+        let runtime = runtimes
+            .iter()
+            .find(|rt| rt.id == runtime_id)
```

Comment:
> What if there are multiple entries for the same runtime ID? Possible in theory when there are multiple versions. 

I guess we would have to use the latest RAK, if we assume that runtimes cannot be downgraded.

> Wouldn't it be more robust if it would search for the target RAK?

I didn't like this solution as testing enclaves all use the same insecure RAK. But maybe this is the way to go, as binding is not checked in that case. Will change.

Reply:
You cannot just use the latest one as that one may not be active yet. I would try matching the target RAK unless there is a security-related downside (currengly I can't think of one).